# Homework 7

## Task 1: Estimating colocalisation between gene expression and disease associations.

Based on the colocalisation tutorial available [here](https://github.com/kauralasoo/MTAT.03.239_Bioinformatics/blob/master/colocalisation/Introduction_to_coloc.md) and using the default prior probabilities of coloc (p1 = p2 = 1e-4, p12 = 1e-5), estimate the colocalisation of genetic associations for the following traits:

 1. Expression of PTK2B gene in the naive condition and Alzheimer's disease (AD)
 2. Expression of ICOSLG gene in the naive condition and ulcerative colitis (UC).
 3. Expression of the TRAF1 gene and rheumatoid arthritis, both in the naive as well as in the IFNg + Salmonella condition.

For all three analyses, report the posterior probabilities for all 5 hypothesis (H0-H4). To remind yourself what the hypotheses  were, have a look at the [original paper](http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004383). 

In all three cases, make Manhattan plots for both the disease and gene expression associations and on both plots, highlight the lead eQTL and 

## Task 2: Understanding the impact of prior probabilities on colocalisation.

## Task 3: How sensitive is colocalisation to prior probabilities? Explore this at the PTK2B and TRAF1 loci.


<!--stackedit_data:
eyJoaXN0b3J5IjpbLTE2OTk5NDA3NjEsLTg3NjE0MTA3NiwyMD
MxMzM3MSwtMTUyNzgyNTAwMV19
-->