# Homework 7

## Task 1: Estimating colocalisation between gene expression and disease associations.

Based on the colocalisation tutorial available [here](https://github.com/kauralasoo/MTAT.03.239_Bioinformatics/blob/master/colocalisation/Introduction_to_coloc.md) and using the default prior probabilites of coloc (p1 = p2 = 1e-4, p12 = 1e-5), estimate the colocalisation of genetic associations for the following traits:

 - Expression of PTK2B gene in the naive condition and Alzheimer's disease (AD).
 - Expression of ICOSLG gene in the naive condition and ulcerative colitis (UC).
 - Expression of the TRAF1 gene and rheumatoid arthritis, both in the naive as well as in the IFNg + Salmonella condition.

For all three analyses, report the posterior probabilities for for all 5 hypothesis. To remind yourself what they were, have a look at the original paper. 

## Task 2: Understanding the impact of prior probabilities on colocalisation.

## Task 3: How sensitive is colocalisation to prior probabilities? Explore this at the PTK2B and TRAF1 loci.


<!--stackedit_data:
eyJoaXN0b3J5IjpbLTYxNDU1Mjk1OCwtODc2MTQxMDc2LDIwMz
EzMzcxLC0xNTI3ODI1MDAxXX0=
-->