---
title: "Introduction to colocalisation"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=FALSE}
suppressMessages(library("dplyr"))
suppressMessages(library("coloc"))
```

## Import the necessary R packages
```{r}
library("dplyr")
library("coloc")
```

## Import association summary statistics
You can find the full eQTL summary statistics from Zenodo: https://zenodo.org/record/1158560. Since the TRAF1 eQTL was specific to the IFNg + Salmonella condition, then the file that you need is RNA_FastQTL_IFNg_SL1344_500kb_pvalues.sorted.txt.gz. For the PTK2B and ICOSLG eQTLs you should use the RNA_FastQTL_naive_500kb_pvalues.sorted.txt.gz file, because those eQTLs were detected in the naive condition. Since the summary files are quite big (~430mb compressed), you might not want to load the complete files into R, especialy because you only need data from specific genes.

The simplest thing to do is to extract the TRAF1 eQTL summary statistics from the file using grep:
```{
gunzip -c  RNA_FastQTL_IFNg_SL1344_500kb_pvalues.sorted.txt.gz | grep ENSG00000056558 > TRAF1_eQTL.txt
```

Now we can just load the eQTL summary statistics into R
```{r}
eQTL_summaries = read.table("TRAF1_eQTL.txt", stringsAsFactors = FALSE, col.names = c("phenotype_id","chr","pos","snp_id","distance","p_nominal","beta")) %>% tbl_df()
eQTL_summaries
```

We can also import the rheumatoid arhtris (RA) GWAS summary statistics from the TRAF1 locus:
```{r TRAF1}
gwas_summaries = read.table("RA_GWAS_TRAF1_locus.txt", stringsAsFactors = FALSE, header = TRUE) %>% tbl_df()
gwas_summaries
```

## Filter the summary statistics to a region +/- 200kb from the lead eQTL variant

Identify the lead eQTL variant (the genetic variant that has the smallest p-value in the eQTL analyis):
```{r}
lead_var = dplyr::arrange(eQTL_summaries, p_nominal) %>% dplyr::filter(row_number() == 1)
```
Keep only those eQTL variants that are +/- 200kb from the lead variant

```{r}
eQTL_filtered = dplyr::filter(eQTL_summaries, (pos > lead_var$pos - 200000) & (pos < lead_var$pos + 200000))
```

Keep only those variants that are present in both eQTL and GWAS datasets:
```{r}
shared_variants = intersect(eQTL_filtered$snp_id, gwas_summaries$snp_id)
eQTL_shared = dplyr::filter(eQTL_filtered, snp_id %in% shared_variants) %>% dplyr::arrange(pos)
gwas_shared = dplyr::filter(gwas_summaries, snp_id %in% shared_variants) %>% dplyr::arrange(pos)
```
Copy minor allele frequency from the GWAS summary statistics into the eQTL summary statistics
```{r}
eQTL_shared = dplyr::left_join(eQTL_shared, dplyr::select(gwas_shared, snp_id, MAF), by = "snp_id")
```

### Test for colocalisation between eQTL and GWAS summary statistics

First, let's set up both datasets for colocalisation. Note that if the `log_OR` column in the GWAS dataset contains only NAs then you should use the `beta` column instead.
```{r}
eQTL_dataset = list(pvalues = eQTL_shared$p_nominal, 
                    N = 84, #The sample size of the eQTL dataset was 84
                    MAF = eQTL_shared$MAF, 
                    type = "quant", 
                    beta = eQTL_shared$beta,
                    snp = eQTL_shared$snp_id)
gwas_dataset = list(beta = gwas_shared$log_OR, #If log_OR column is full of NAs then use beta column instead
                    varbeta = gwas_shared$se^2, 
                    type = "cc", 
                    snp = gwas_shared$snp_id,
                    s = 0.5, #This is acutally not used, because we already specified varbeta above.
                    MAF = gwas_shared$MAF)
```

Now we can finally run coloc itself
```{r}
coloc_res = coloc::coloc.abf(dataset1 = eQTL_dataset, dataset2 = gwas_dataset,p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)
coloc_res$summary
```

